### ISSUE TITLE
experimental_pollInterval doesn't continue to poll if a federated service is not reachable at startup

### ISSUE THREAD

**Issue by guillaumeNaimi** [apollographql/federation#235]
**Package**
@apollo/gateway 0.21.0

**Latest version where the problem did not occur**
@apollo/gateway 0.14.0

**Expected behavior**
If a federated service in gateway service list is not reachable when you launch the gateway, the gateway will retry to reach the service to get its schema after the poll interval set in configuration.

**Actual behavior**
If a federated service in gateway service list is not reachable when you launch the gateway, the gateway will stop the polling so the gateway will never retry to fetch the federated service schema and always return a 500 error.
The behavior happens when there is these errors on logs at startup : 
`Error checking for changes to service definitions: Couldn't load service definitions for [federated_service_name] at [federated_server_url] request to [federated_service_url] failed, reason: connect ECONNREFUSED 127.0.0.1:6000
This data graph is missing a valid configuration. Couldn't load service definitions for [federated_service_name] at  [federated_service_url]: request to  [federated_service_url] failed, reason: connect ECONNREFUSED 127.0.0.1:6000`

**Steps to reproduce**
To reproduce this behavior, you can : 
- Launch first the gateway with a simple federated service in service list and the `experimental_pollInterval` set.
- The error above will be displayed in the logs.
- After the gateway is launch, launch the federated service you set in service list. 
- Even if the federated service is available, the gateway will not try to refetch its schema after the interval set in the gateway.
- If you start the federated service before the gateway, shutdown the federated service, wait to have the above error in the gateway logs and restart the federated service, it will work as expected and refetch the schema at each interval you set.

**Comment by ruanmartinelli** [comment]
We had the same problem. I built a workaround that health-checks each subservice before starting the gateway. You can make a request to `<host>/.well-known/apollo/server-health` to know the service is ready or not. 

A `wait-for-it` script [[1](https://github.com/vishnubob/wait-for-it), [2](https://github.com/bash-suite/wait-host)] can also be an option depending on your workflow.

**Comment by flippidippi** [comment]
Having the same issue here. It should behave the same way as it does after startup and keep retrying to get the schema if it fails at start.

**Comment by glasser** [comment]
Hi! I think this is a duplicate of #335. I am planning to work on #335 very soon.

My intended solution is different than yours. I do think there is generally a distinction between "can't load the schema on startup", which likely indicates a configuration issue, and "we could load it before but can't load it now", which likely indicates a temporary network issue.  So we will stick to a design where a failure to load the schema on startup is a permanent error. However, we'll actually provide a way of waiting for and handling that error.

Specifically, we'll introduce `async ApolloServer.start()`.  This can be an optional method in AS2 (future major versions may make it required) that is highly recommended for users of `@apollo/gateway`.  The recommended way of starting your server will be to `await server.start()` before your HTTP server starts to listen; errors loading the initial schema can cause an immediate failure that your app can handle.  (If you want your app to be resilient to being unable to load the schema on startup, you can just wrap your `const s = new ApolloServer(...); await s.start()` in a retry loop that fits your use case.)
